Tests/MongoSwiftTests/ReadPreferenceTests.swift

```diff
@@ -0,0 +1,89 @@
+
+        let primaryPreferred = ReadPreference(.primaryPreferred)
+        expect(primaryPreferred.mode).to(equal(.primaryPreferred))
+
+        let secondary = ReadPreference(.secondary)
+        expect(secondary.mode).to(equal(.secondary))
+
+        let secondaryPreferred = ReadPreference(.secondaryPreferred)
+        expect(secondaryPreferred.mode).to(equal(.secondaryPreferred))
+
+        let nearest = ReadPreference(.nearest)
+        expect(nearest.mode).to(equal(.nearest))
+    }
+
+    func testReadPreferenceTagSets() throws {
+        let rpNoTagSets = try ReadPreference(.nearest, tagSets: nil)
+        expect(rpNoTagSets.tagSets).to(equal([]))
+
+        let rpSomeTagSets = try ReadPreference(.nearest, tagSets: [["dc": "east"], []])
+        expect(rpSomeTagSets.tagSets).to(equal([["dc": "east"], []]))
+
+        let rpOnlyEmptyTagSet = try ReadPreference(.nearest, tagSets: [[]])
+        expect(rpOnlyEmptyTagSet.tagSets).to(equal([[]]))
+
+        // Non-empty tag sets cannot be combined with primary mode
+        expect(try ReadPreference(.primary, tagSets: [["dc": "east"], []])).to(throwError())
+        expect(try ReadPreference(.primary, tagSets: [[]])).to(throwError())
+    }
+
+    func testReadPreferenceMaxStalenessSeconds() throws {
+        let rpNoMaxStaleness = try ReadPreference(.nearest, maxStalenessSeconds: nil)
+        expect(rpNoMaxStaleness.maxStalenessSeconds).to(beNil())
+
+        let rpMinMaxStaleness = try ReadPreference(.nearest, maxStalenessSeconds: 90)
+        expect(rpMinMaxStaleness.maxStalenessSeconds).to(equal(90))
+
+        let rpLargeMaxStaleness = try ReadPreference(.nearest, maxStalenessSeconds: 2147483647)
+        expect(rpLargeMaxStaleness.maxStalenessSeconds).to(equal(2147483647))
+
+        // maxStalenessSeconds cannot be less than 90
+        expect(try ReadPreference(.nearest, maxStalenessSeconds: -1)).to(throwError())
+        expect(try ReadPreference(.nearest, maxStalenessSeconds: 0)).to(throwError())
+        expect(try ReadPreference(.nearest, maxStalenessSeconds: 89)).to(throwError())
+    }
+
+    func testEquatable() {
+        expect(ReadPreference(.primary)).to(equal(ReadPreference(.primary)))
+        expect(ReadPreference(.primary)).toNot(equal(ReadPreference(.primaryPreferred)))
+        expect(ReadPreference(.primary)).toNot(equal(ReadPreference(.secondary)))
+        expect(ReadPreference(.primary)).toNot(equal(ReadPreference(.secondaryPreferred)))
+        expect(ReadPreference(.primary)).toNot(equal(ReadPreference(.nearest)))
+
+        expect(try ReadPreference(.secondary, tagSets: nil))
+            .to(equal(ReadPreference(.secondary)))
+        expect(try ReadPreference(.secondary, tagSets: []))
+            .to(equal(try! ReadPreference(.secondary, tagSets: [])))
+        expect(try ReadPreference(.secondary, tagSets: [["dc": "east"], []]))
+            .to(equal(try! ReadPreference(.secondary, tagSets: [["dc": "east"], []])))
```

Comment:
I think it makes sense to `try!` here, since I should guarantee that the expected value is well-formed. If it wasn't, I'd prefer a loud error rather than simply a non-equal comparison.

Do you agree this is preferable to `try` and `try?`, the latter of which might end up comparing against an expected `nil`?